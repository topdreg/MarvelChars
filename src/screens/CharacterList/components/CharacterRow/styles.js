import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    width: '100%',
    height: 100,
    borderBottomWidth: 0.4,
  },
  image: {
    height: 70,
    width: 70,
  },
  pressable: {
    width: '100%',
    height: '100%',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingLeft: '5%',
    paddingRight: '5%',
  },
});
