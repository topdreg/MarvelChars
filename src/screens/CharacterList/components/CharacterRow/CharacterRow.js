import React from 'react';
import {Navigation} from 'react-native-navigation';
import {View, Image, Text, TouchableOpacity} from 'react-native';
import styles from './styles';

const CharacterRow = (props) => {
  const {characterData, style} = props;
  const {name, thumbnail} = characterData;
  const imageUri = `${thumbnail.path}.${thumbnail.extension}`;
  return (
    <View style={[style, styles.container]}>
      <TouchableOpacity
        style={styles.pressable}
        onPress={() => toCharacterPage(props, imageUri)}>
        <Text>{name}</Text>
        <Image style={styles.image} source={{uri: imageUri}} />
      </TouchableOpacity>
    </View>
  );
};

const toCharacterPage = async (props, imageUri) => {
  const {characterData, componentId} = props;
  await Navigation.push(componentId, {
    component: {
      name: 'CharacterPage',
      passProps: {
        characterData,
        imageUri,
      },
    },
  });
};

export default CharacterRow;
