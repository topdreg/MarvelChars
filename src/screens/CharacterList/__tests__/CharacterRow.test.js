import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import CharacterRow from '../components/CharacterRow';

it('renders correctly', () => {
  const characterData = {
    name: 'A-Bomb',
    thumbnail: 'http://i.annihil.us/u/prod/marvel/i/mg/3/20/5232158de5b16',
  };
  const tree = renderer
    .create(<CharacterRow characterData={characterData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
