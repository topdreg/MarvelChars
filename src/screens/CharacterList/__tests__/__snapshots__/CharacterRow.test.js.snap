// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "borderBottomWidth": 0.4,
        "height": 100,
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingLeft": "5%",
        "paddingRight": "5%",
        "width": "100%",
      }
    }
  >
    <Text>
      A-Bomb
    </Text>
    <Image
      source={
        Object {
          "uri": "undefined.undefined",
        }
      }
      style={
        Object {
          "height": 70,
          "width": 70,
        }
      }
    />
  </View>
</View>
`;
